app/controllers/spree/admin/reports_controller_decorator.rb

```diff
@@ -181,20 +189,37 @@ def sales_tax
   end
 
   def bulk_coop
-    # -- Prepare date parameters
-    prepare_date_params params
+    if request.format.json?
```

Comment:
I do not see a problem reading `if - else` conditions. Extracting all actions and creating another controller with the same actions in my opinion is an overkill. Why can't the same controller serve both types if it is able to do that? 

Replacing `if - else` with `respond_to` probably would be more elegant way, but that can be done in the next step of this feature.